src/index/CompressedBlockPrefiltering.cpp

```diff
@@ -0,0 +1,170 @@
+//  Copyright 2024, University of Freiburg,
+//                  Chair of Algorithms and Data Structures
+//  Author: Hannes Baumann <baumannh@informatik.uni-freiburg.de>
+
+#include "index/CompressedBlockPrefiltering.h"
+
+namespace prefilterExpressions {
+
+// SECTION RELATIONAL OPERATIONS
+//______________________________________________________________________________
+template <CompOp Comparison>
+bool RelationalExpressions<Comparison>::compareImpl(
+    const ValueId& firstId, const ValueId& secondId) const {
+  using enum CompOp;
+  // For compactness we combine LE and GT here. In evaluate (see below), for
+  // LE we select all blocks at a smaller (lower_bound) index position, and
+  // for GT all the respective blocks at larger or equal indices.
+  if constexpr (Comparison == LE || Comparison == GT) {
+    return compareIds(firstId, secondId, LE) ==
+           valueIdComparators::ComparisonResult::True;
+  } else {
+    // Follows the same logic for evaluation as explained above.
+    // Used for CompOp::LT, CompOp::GE, CompOp::EQ and CompOp::NE.
+    // For CompOp::EQ we also use the LT operation: This is possible because in
+    // evaluate we calculate a lower_bound and upper_bound over this LT
+    // operation, and those bounds act as the plausible range for equality
+    // (evaluation of CompOp::NE follows a similar approach).
+    return compareIds(firstId, secondId, LT) ==
+           valueIdComparators::ComparisonResult::True;
+  }
+}
+
+//______________________________________________________________________________
```

Comment:
I will review the rest of the `RelationalExpressions` once you have resolved the above issue.